Add appveyor build

This would also demonstrate how to install/build this on windows, besides verifying it actually builds on that platform. And hopefully produce proper windows installation instructions because the majority of tickets in this project are related to windows installation fails.

I can make a pull request for this but some pointers would be useful, I read `system32` needs to be on the path for example?

Here's an example of appveyor failing to install node-opus: https://ci.appveyor.com/project/thijstriemstra/videojs-record/build/1.0.451

```
..\..\deps\opus\silk\float\levinsondurbin_FLP.c(29): fatal error C1083: Cannot open include file: 'config.h': No such file or directory [C:\projects\videojs-record\node_modules\node-opus\build\deps\libopus.vcxproj]
  scale_copy_vector_FLP.c
..\..\deps\opus\silk\float\pitch_analysis_core_FLP.c(29): fatal error C1083: Cannot open include file: 'config.h': No such file or directory [C:\projects\videojs-record\node_modules\node-opus\build\deps\libopus.vcxproj]
  scale_vector_FLP.c
..\..\deps\opus\silk\float\scale_copy_vector_FLP.c(29): fatal error C1083: Cannot open include file: 'config.h': No such file or directory [C:\projects\videojs-record\node_modules\node-opus\build\deps\libopus.vcxproj]
  schur_FLP.c
..\..\deps\opus\silk\float\scale_vector_FLP.c(29): fatal error C1083: Cannot open include file: 'config.h': No such file or directory [C:\projects\videojs-record\node_modules\node-opus\build\deps\libopus.vcxproj]
  sort_FLP.c
..\..\deps\opus\silk\float\schur_FLP.c(29): fatal error C1083: Cannot open include file: 'config.h': No such file or directory [C:\projects\videojs-record\node_modules\node-opus\build\deps\libopus.vcxproj]
..\..\deps\opus\silk\float\sort_FLP.c(29): fatal error C1083: Cannot open include file: 'config.h': No such file or directory [C:\projects\videojs-record\node_modules\node-opus\build\deps\libopus.vcxproj]
gyp ERR! build error 
gyp ERR! stack Error: `msbuild` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\appveyor\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\build.js:262:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Windows_NT 6.3.9600
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\projects\videojs-record\node_modules\node-opus
gyp ERR! node -v v8.12.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules\chokidar\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"ia32"})
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-opus@0.3.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-opus@0.3.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\appveyor\AppData\Roaming\npm-cache\_logs\2018-09-16T01_22_51_677Z-debug.log
```

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -1,0 +1,33 @@
+environment:
+  matrix:
+    - nodejs_version: Current
+      platform: x64
+
+install:
+  - ps: Install-Product node $env:nodejs_version $env:platform
+  - set CI=true
+  - npm install --global npm@latest
+  - set PATH=%APPDATA%\npm;%PATH%
+  - npm install
+
+matrix:
+  fast_finish: true
+
+platform:
+  - x64
+
+# Disable automatic builds
+build: off
+
+# Do not build on gh tags
+skip_tags: true
+shallow_clone: true
+
+test_script:
+  - node --version
+  - npm --version
+  - npm test
+
+cache:
+  - '%APPDATA%\npm-cache'
+  - node_modules -> package.json

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -9,6 +9,8 @@ install:
   - npm install --global npm@latest
   - set PATH=%APPDATA%\npm;%PATH%
   - npm install
+  - npm install -g gulp istanbul codeclimate-test-reporter
+  - npm install ogg-packet
 
 matrix:
   fast_finish: true

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -15,9 +15,6 @@ install:
 matrix:
   fast_finish: true
 
-platform:
-  - x64
-
 # Disable automatic builds
 build: off
 

